### PR TITLE
Fix bug #888

### PR DESCRIPTION
**Pull request by aleister09**
This PR fixes issue #888 and I found another issue in favorites and photos(Filter Sections) and I fixed it

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 1]
```diff
@@ -250,10 +250,6 @@ public void run() {
     public void onResume() {
         super.onResume();
 
-        if (remoteOperationAsyncTask != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java]: This IHMO shouldn't be here. Why do we remove it? Async tasks should always be cancelled - thought I'd rather have them cancelled in onPause than onResume.

What does this fixes?

Comment (aleister09) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java]: This code cancel the Async Task after it is started and when the remote operation finish the async task asks if it was canceled and can't set the response
```
protected Object doInBackground(Object[] params) {
                if (getContext() != null && !isCancelled()) {
                    RemoteOperationResult remoteOperationResult = remoteOperation.execute(currentAccount, getContext());

                    if (remoteOperationResult.isSuccess() && remoteOperationResult.getData() != null
                            && !isCancelled() && searchFragment) {
                        mAdapter.setData(remoteOperationResult.getData(), currentSearchType);
                    }

                    return remoteOperationResult.isSuccess();
                } else {
                    return false;
                }
            }
```

**Comment by mario** [comment]
@tobiasKaminsky well, it certainly solved the problem, but is it the right fix? You decide...

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 1 above)
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java]: This is somehow outdated as I have changed this again in https://github.com/nextcloud/android/pull/896
So we can ignore this here.
:+1: for the rest

**Comment by mario** [comment]
@tobiasKaminsky so you want to approve as-is? No changes needed? We can just merge?

**Comment by tobiasKaminsky** [comment]
I reverted this as it might get confusing.
So this PR is only about #888

**Comment by tobiasKaminsky** [comment]
:+1:

**Comment by mario** [comment]
👍

**Pull request merged**